File: Backend/Soxapi/app/Models/Mercancia.php
<?php

namespace App\Models;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Mercancia extends Model
{
    use HasFactory;
    
    protected $table = 'mercancias';

    protected $fillable = [
        'proveedores_id',
        'usuarios_id',
        'fecha_ingreso',
        'numero_remesa',
        'origen_mercancia',
        'destino_mercancia',
        'nombre_remitente',
        'documento_remitente',
        'direccion_remitente',
        'celular_remitente',
        'nombre_destinatario',
        'documento_destinatario',
        'direccion_destinatario',
        'celular_destinatario',
        'valor_declarado',
        'valor_flete',
        'valor_total',
        'peso',
        'unidades',
        'observaciones',
        'tipo_pago_id'
        
    ];
    
    public function proveedores()
    {
        return $this->belongsTo(Proveedor::class, 'proveedores_id');
    }

    public function usuarios()
    {
        return $this->belongsTo(User::class, 'usuarios_id');
    }

    public function tipopago()
    {
        return $this->belongsTo(TipoPago::class, 'tipo_pago_id');
    }

    public function despachos()
    {
        return $this->hasMany(Despacho::class, 'mercancias_id');
    }

    public function entregas()
    {
        return $this->hasMany(Entrega::class, 'mercancias_id');
    }

    public function devoluciones()
    {
        return $this->hasMany(Devolucion::class, 'mercancias_id');
    }
    public function seguimientos()
{
    return $this->hasMany(Seguimiento::class, 'mercancias_id');
}
protected static function booted()
{
    static::created(function ($mercancia) {
        $mercancia->seguimientos()->create([
            'estado' => 'Ingresado a bodega',
        ]);
    });
}



}

File: TRIMESTRE 8/AppMovil-Kotlin/src/main/kotlin/com/example/Sox/mercancias/Mercancia.kt
package com.example.Sox.mercancias

import java.time.LocalDate

data class Mercancia(
    val id: Int? = null,
    val proveedores_id: Int,
    val usuarios_id: Int,
    val fecha_ingreso: LocalDate,
    val numero_remesa: String,
    val origen_mercancia: String,
    val destino_mercancia: String,

    val nombre_remitente: String,
    val documento_remitente: String,
    val direccion_remitente: String,
    val celular_remitente: String,

    val nombre_destinatario: String,
    val documento_destinatario: String,
    val direccion_destinatario: String,
    val celular_destinatario: String,

    val valor_declarado: Double,
    val valor_flete: Double,
    val valor_total: Double,
    val peso: Double,
    val unidades: Int,
    val observaciones: String? = null,

    val tipo_pago_id: Int
)
